src/ansys/fluent/core/utils/file_transfer_service.py

```diff
@@ -74,6 +83,8 @@ def upload(self, file_name: str, remote_file_name: Optional[str] = None):
                 self.file_service.upload_file(expanded_file_path, upload_file_name)
             else:
                 raise FileNotFoundError(f"{file_name} does not exist.")
+        elif not pypim.is_configured():
+            raise PyPIMConfigurationError()
```

Comment:
I believe we should check if PyPIM is configured first, before the other steps. It doesn't make much sense to me to go ahead with the `file_service` steps if PyPIM is not configured to begin with. Thoughts @hpohekar ?

Reply:
Yes, I thought earlier similar to this but I also remembered 1 comment that I received similar to this scenario and I followed that one.

![image](https://github.com/ansys/pyfluent/assets/106588300/2fb1af5e-7379-40a0-8528-571e846f10da)

Reply:
It's updated now. Thank you.